### ISSUE TITLE
Update vendor dependencies to support FIPS on older Python versions

### ISSUE THREAD

**Issue by andrewwongscale** [python-poetry/poetry#8223]
[x] I have searched the [issues](https://github.com/python-poetry/poetry/issues) of this repo and believe that this is not a duplicate.
[x] I have searched the [FAQ](https://python-poetry.org/docs/faq/) and general [documentation](https://python-poetry.org/docs/) and believe that my question is not already covered.

## Feature Request

Hello,

Would it be possible to update the vendor dependencies (like this [PR](https://github.com/python-poetry/poetry-core/pull/503)), particularly lark to capture [this update](https://github.com/lark-parser/lark/issues/1250) from February 2023?

I'm running into [this exact issue](https://github.com/lark-parser/lark/issues/1250) from lark on `poetry install`, where I'm unable to build because Poetry's version of Lark is using md5 hashing which is blocked on FIPS systems.

Similar ask from last year: https://github.com/python-poetry/poetry/issues/4986. This fixed it for Python versions >= 3.9. Unfortunately, our work is on Python 3.8

**Comment by radoering** [comment]
We'll do an update of the vendored dependencies just before the next poetry-core release (which might be in August).
